src/predicates/intersects.jl

```diff
@@ -106,22 +107,94 @@ function intersects(g₁::Geometry{Dim,T}, g₂::Geometry{Dim,T}) where {Dim,T}
       AB = B - A
       AO = O - A
       d = AB × AO × AB
-    else # simplex case
-      C, B, A = points
-      AB = B - A
-      AC = C - A
-      AO = O - A
-      ABᵀ = AC × AB × AB
-      ACᵀ = AB × AC × AC
-      if ABᵀ ⋅ AO > zero(T)
-        popat!(points, 1) # pop C
-        d = ABᵀ
-      elseif ACᵀ ⋅ AO > zero(T)
-        popat!(points, 2) # pop B
-        d = ACᵀ
-      else
-        return true
-      end
+    else
+      d = gjk!(Val(Dim), O, points)
+      isnothing(d) && return true
+    end
+  end
+end
+
+"""
+    gjk!(::Val{Dim}, origin, points) where Dim
+
+Perform one iteration of the GJK algorithm.
+
+It returns `nothing` if the `Dim`-simplex represented by `points`
+contains the origin point. Otherwise, it returns a vector with
+the direction for searching the next point.
+
+If the simplex is complete, it removes one point from the set to
+make room for the next point. A complete simplex must have `Dim + 1` points. 
+
+See also [`intersects`](@ref).
+"""
+function gjk! end
+
+function gjk!(::Val{2}, origin, points)
+  @assert length(points) == 3
+  # triangle simplex case
+  C, B, A = points
+  AB = B - A
+  AC = C - A
+  AO = origin - A
+  ABᵀ = AC × AB × AB
+  ACᵀ = AB × AC × AC
+  T = coordtype(origin)
+  if ABᵀ ⋅ AO > zero(T)
+    popat!(points, 1) # pop C
+    return ABᵀ
+  elseif ACᵀ ⋅ AO > zero(T)
+    popat!(points, 2) # pop B
+    return ACᵀ
+  else
+    return nothing
+  end
+end
+
+function gjk!(::Val{3}, origin, points)
+  if length(points) == 3
+    # triangle case
+    C, B, A = points
+    AB = B - A
+    AC = C - A
+    AO = origin - A
+    ABCᵀ = AB × AC
+    if ABCᵀ ⋅ AO < 0
+      points[1], points[2] = points[2], points[1]
+      ABCᵀ = -ABCᵀ
+    end
+    return ABCᵀ
+  else
+    # tetrahedron simplex case
+    #      A
+    #    / | \
+    #   /  D  \
+    #  / /   \ \
+    # B ------- C
+    # Simplex faces (with normal vectors pointing to the centroid):
+    # ABC, ADB, BDC, ACD
+    # (AXY = AX × AY)
+    # ABC normal vector points to vertex D
+    D, C, B, A = points
+    AB = B - A
+    AC = C - A
+    AD = D - A
+    AO = origin - A
+    ABCᵀ = AB × AC
+    ADBᵀ = AD × AB
+    ACDᵀ = AC × AD
+    T = coordtype(origin)
+    if ABCᵀ ⋅ AO > zero(T)
+      popat!(points, 1) # pop D
+      return ABCᵀ
```

Comment:
We have JuliaFormatter set on this repository. It will fix most of the code style issues for us automatically. The `return` issue is something we can't fix automatically, but it is easy to share with first-time contributors. Thanks for trying to adhere to it.

No problem if you need to change the `minkowskipoint` and `minkowskiorigin`.